spring-web/src/main/java/org/springframework/http/RequestEntity.java

```diff
@@ -146,14 +157,31 @@ public RequestEntity(@Nullable T body, @Nullable MultiValueMap<String, String> h
 	 * @since 4.3
 	 */
 	public RequestEntity(@Nullable T body, @Nullable MultiValueMap<String, String> headers,
-			@Nullable HttpMethod method, URI url, @Nullable Type type) {
+						 @Nullable HttpMethod method, URI url, @Nullable Type type) {
 
 		super(body, headers);
 		this.method = method;
 		this.url = url;
 		this.type = type;
 	}
 
+	/**
+	 * Private Constructor with method, URL, UriTemplate and varargs urivariables but without body nor headers.
+	 * @param method the method
+	 * @param url the URL
+	 * @param uri the UriTemplate
+	 * @param uriVariables the uriVariables
+	 */
+	private RequestEntity(MultiValueMap<String, String> headers, HttpMethod method, URI url,
+						  String uri, Object... uriVariables) {
+		super(null, headers);
+		this.method = method;
+		this.url = url;
+		this.type = null;
+		this.uri = uri;
+		this.uriVariables = uriVariables;
+	}
```

Comment:
The `url`, `uri`, and `uriVariables` arguments should be `@Nullable` and `org.springframework.util.Assert` should be used to check that either `url` or `uri` is provided. Otherwise the `getUrl` methods below would fail.